### ISSUE TITLE
strange sha512 integrity error in prod

### ISSUE THREAD

**Issue by vladikoff** [mozilla/fxa-content-server#4187]
### Reported issue:

<img width="832" alt="ae" src="https://cloud.githubusercontent.com/assets/128755/18801241/96a77bf0-81af-11e6-9fac-e8f6a860935d.png">
<img width="1102" alt="eaeqweqwe" src="https://cloud.githubusercontent.com/assets/128755/18801534/331a07fe-81b1-11e6-9728-3306f25f05f7.png">

### My CSS size is different: 

![image](https://cloud.githubusercontent.com/assets/128755/18801547/52d5d8e8-81b1-11e6-985d-03cbfea5058f.png)

```
[16:41:52]  <mkaply>	I'm curious why you get a different Firefox accounts signup page if your accept language is zh-TW - http://imgur.com/a/vGeFl

[16:42:17]  <vladikoff>	mkaply, looks like CSS failed to load, cc @jrgm 
[16:42:41]  <vladikoff>	mkaply , try refreshing, also firefox devtool's network tab and screenshot that it keeps happening
[16:43:04]  <mkaply>	vladikoff: I tried all that. No change. You can easily recreate by just change the accept language in a US build to zh-TW
[16:43:11]  <vladikoff>	mkaply, checking
[16:45:20]  <vladikoff>	mkaply, i changed `intl.accept_languages`  http://v14d.com/i/57e594477f8b0.png  is that what you did?
[16:46:16]  <mkaply>	vladikoff: YEp. intl.accept_languages;zh-TW
[16:46:36]  <vladikoff>	mkaply, works fine for me
[16:46:57] 	gavin (gavin@irc.mozilla.org) joined the channel
[16:46:57]  <vladikoff>	mkaply , can you check what's going on in firefox devtools?
[16:47:01]  <mkaply>	OK. I'll see if I can figure out why the CSS isn't loading. Weird.
[16:47:21]  <vladikoff>	mkaply, can you open https://accounts.cdn.mozilla.net/styles/f8ac2a03.zh_TW.css
[16:48:52]  <mkaply>	yep. Posting my network info now
[16:49:51]  <mkaply>	I wish there was a simple pastebin for images.
[16:51:03]  <mkaply>	http://picpaste.com/pics/foopy-iOoU7h8k.1474663854.png
[16:51:29]  <vladikoff>	mkaply, any errors in 'Console' ?
[16:52:38]  <mkaply>	Only thing that even remotely looks like an error is None of the “sha512” hashes in the integrity attribute match the content of the subresource.
[16:53:00]  <mkaply>	Interesting that my CSS is 10.19 and yours is 10.18
[16:54:07]  <mkaply>	main.js is a different size as well.
[16:55:22]  <mkaply>	Caching issue maybe?
[16:55:50]  <vladikoff>	mkaply, interesting!
[16:56:17]  <vladikoff>	mkaply, can you take a pic of `None of the “sha512” hashes` error?
[16:56:52]  <vladikoff>	jrgm, around?
[16:57:32]  <jrgm>	vladikoff: hmm
[16:57:58]  <mkaply>	http://www.picpaste.com/pics/foopy2-EzQvnIIF.1474664267.png
[16:58:00]  <vladikoff>	jrgm, do we have node 4 and 0.10 boxes live? do you think the page served can be different and ask for a different css?
[16:58:15]  <mkaply>	FYI, I've tried two different machines, and a fresh profile (on a zh-TW build)
[16:59:22]  <jrgm>	vladikoff: not live. But I did push content into CDN, and had already noticed this sri issue on the (not-live) node 4 stack. 
[16:59:54]  <vladikoff>	jrgm, any ideas why is it only zh-TW or some specific case?
[17:00:30]  <jrgm>	Nope
[17:02:15]  <vladikoff>	mkaply, thanks for the screenshot
[17:02:32]  <mkaply>	vladikoff: No problem. Is that sha512 error the source of the problem?
[17:03:02]  <mkaply>	Let me know if there is anything else you want me to test.
[17:04:28]  <vladikoff>	mkaply, for testing purposes you can disable: `security.sri.enable` in about:config and see it loads. MAke sure to reenable after, it is a security setting
```

**Comment by vladikoff** [comment]
```
[17:04:28]  <vladikoff>	mkaply, for testing purposes you can disable: `security.sri.enable` in about:config and see it loads. MAke sure to reenable after, it is a security setting
[17:12:54]  <mkaply>	Yep That fixed it
```

**Comment by vladikoff** [comment]
This was fixed: 

```
[17:37:56]  <jbuck>	think about it though... if we have an html file inclding a css files referencing an image
[17:38:04]  <jbuck>	we'd need to sri-hash 3 times to get the correct hash right?
[17:38:18]  <jbuck>	is my thinking correct?
[17:40:26]  <jbuck>	mkaply: okay, cdn cache is cleared, working for me here. Can you verify it's working for you now? 
```

cc @jbuck @shane-tomlinson

**Comment by jbuck** [comment]
The differences between the two files was the cache-busting code in each image url:

```patch
$ diff --normal old.css new.css
657c657
<     background-image: url(/images/4a7891df.firefox.png);
---
>     background-image: url(/images/127caab3.firefox.png);
677c677
<         background-image: url(/images/e5686dd1.mozilla.png);
---
>         background-image: url(/images/b6ae763b.mozilla.png);
718c718
<     background-image: url(/images/500ea6d8.choose_what_to_sync_devices.jpg);
---
>     background-image: url(/images/2c8a1da5.choose_what_to_sync_devices.jpg);
729c729
<         background-image: url(/images/87952039.firefox@2x.png)
---
>         background-image: url(/images/f5f9c238.firefox@2x.png)
732c732
<         background-image: url(/images/58d51824.choose_what_to_sync_devices@2x.jpg)
---
>         background-image: url(/images/3b0ef411.choose_what_to_sync_devices@2x.jpg)
1568c1568
<     background-image: url(/images/4a7891df.firefox.png);
---
>     background-image: url(/images/127caab3.firefox.png);
1578c1578
<         background-image: url(/images/87952039.firefox@2x.png)
---
>         background-image: url(/images/f5f9c238.firefox@2x.png)
2520c2520
<     background-image: url(/images/994d7b9f.graphic_mail.png);
---
>     background-image: url(/images/1a6fdeab.graphic_mail.png);
2530c2530
<         background-image: url(/images/d6ca0d85.graphic_mail@2x.png)
---
>         background-image: url(/images/fb0b1883.graphic_mail@2x.png)
2579c2579
<     background-image: url(/images/b281253c.android_logo.png);
---
>     background-image: url(/images/8ecc4030.android_logo.png);
2591c2591
<         background-image: url(/images/5bf5b791.android_logo@2x.png)
---
>         background-image: url(/images/e735e1d8.android_logo@2x.png)
```

So it's also possible that this is fixed by fixing #4180 but I noticed that @jrgm filed issue number 12 in the private issue that may require us to sri-hash 3x to fix this correctly.

**Comment by jrgm** [comment]
Confirming that difference and noting it's for images **and** font files.

But this points out that the checksum of all the css files is not a checksum (correct md5 or not) of the final css file content, but a checksum of presumably some earlier form of the css content.

It also points out that functional tests do not have a test that confirms that css has actually loaded.
